src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -262,34 +260,23 @@ def list_role_assignments(cmd,  # pylint: disable=too-many-locals, too-many-bran
 
     # Fill in role definition names
     if fill_role_definition_name:
-        worker = MultiAPIAdaptor(cmd.cli_ctx)
         role_defs = list(definitions_client.list(
             scope=scope or ('/subscriptions/' + definitions_client._config.subscription_id)))
-        role_dics = {i.id: worker.get_role_property(i, 'role_name') for i in role_defs}
-        for i in results:
-            if not i.get('roleDefinitionName'):
-                if role_dics.get(worker.get_role_property(i, 'roleDefinitionId')):
-                    worker.set_role_property(i, 'roleDefinitionName',
-                                             role_dics[worker.get_role_property(i, 'roleDefinitionId')])
-                else:
-                    i['roleDefinitionName'] = None  # the role definition might have been deleted
+        role_dics = {rd.id: rd.role_name for rd in role_defs}
+        for ra in results:
+            ra['roleDefinitionName'] = role_dics.get(ra['roleDefinitionId'])
 
     # Fill in principal names
     if fill_principal_name:
-        worker = MultiAPIAdaptor(cmd.cli_ctx)
-        principal_ids = set(worker.get_role_property(i, 'principalId')
-                            for i in results if worker.get_role_property(i, 'principalId'))
+        principal_ids = set(ra['principalId'] for ra in results)
 
         if principal_ids:
             try:
                 principals = _get_object_stubs(graph_client, principal_ids)
                 principal_dics = {i[ID]: _get_displayable_name(i) for i in principals}
 
-                for i in [r for r in results if not r.get('principalName')]:
```

Comment:
It is not possible for a `roleAssignment` object to have `principalName` property: https://learn.microsoft.com/en-us/rest/api/authorization/role-assignments/list-for-scope, so this statement is unnecessary.